otj-metrics-core/src/main/java/com/opentable/metrics/OTInstrumentedHandler.java

```diff
@@ -0,0 +1,343 @@
+        this.connectRequests = metricRegistry.timer(name(prefix, "connect-requests"));
+        this.moveRequests = metricRegistry.timer(name(prefix, "move-requests"));
+        this.otherRequests = metricRegistry.timer(name(prefix, "other-requests"));
+
+        metricRegistry.register(name(prefix, "percent-4xx-1m"), new RatioGauge() {
+            @Override
+            protected Ratio getRatio() {
+                return Ratio.of(responses[3].getOneMinuteRate(),
+                        requests.getOneMinuteRate());
+            }
+        });
+
+        metricRegistry.register(name(prefix, "percent-4xx-5m"), new RatioGauge() {
+            @Override
+            protected Ratio getRatio() {
+                return Ratio.of(responses[3].getFiveMinuteRate(),
+                        requests.getFiveMinuteRate());
+            }
+        });
+
+        metricRegistry.register(name(prefix, "percent-4xx-15m"), new RatioGauge() {
+            @Override
+            protected Ratio getRatio() {
+                return Ratio.of(responses[3].getFifteenMinuteRate(),
+                        requests.getFifteenMinuteRate());
+            }
+        });
+
+        metricRegistry.register(name(prefix, "percent-5xx-1m"), new RatioGauge() {
+            @Override
+            protected Ratio getRatio() {
+                return Ratio.of(responses[4].getOneMinuteRate(),
+                        requests.getOneMinuteRate());
+            }
+        });
+
+        metricRegistry.register(name(prefix, "percent-5xx-5m"), new RatioGauge() {
+            @Override
+            protected Ratio getRatio() {
+                return Ratio.of(responses[4].getFiveMinuteRate(),
+                        requests.getFiveMinuteRate());
+            }
+        });
+
+        metricRegistry.register(name(prefix, "percent-5xx-15m"), new RatioGauge() {
+            @Override
+            public Ratio getRatio() {
+                return Ratio.of(responses[4].getFifteenMinuteRate(),
+                        requests.getFifteenMinuteRate());
+            }
+        });
+
+        this.listener = new AsyncListener() {
+            @Override
+            public void onTimeout(AsyncEvent event) throws IOException {
+                asyncTimeouts.mark();
+            }
+
+            @Override
+            public void onStartAsync(AsyncEvent event) throws IOException {
+                event.getAsyncContext().addListener(this);
+            }
+
+            @Override
+            public void onError(AsyncEvent event) throws IOException {
+            }
+
+            @Override
+            public void onComplete(AsyncEvent event) throws IOException {
+                final HttpChannelState state = ((AsyncContextEvent)event).getHttpChannelState();
+
+                final Request request = state.getBaseRequest();
+                final long elapsed = System.currentTimeMillis() - request.getTimeStamp();
```

Comment:
See comment below about measuring elapsed time.

Reply:
This was taken directly from DropWizard Metrics. Let me post a diff to show what was changed.